packages/mobile/src/recipients/recipient.ts

```diff
@@ -2,67 +2,76 @@ import { parsePhoneNumber } from '@celo/utils/lib/phoneNumbers'
 import * as fuzzysort from 'fuzzysort'
 import { TFunction } from 'i18next'
 import { MinimalContact } from 'react-native-contacts'
-import { AddressToE164NumberType, E164NumberToAddressType } from 'src/identity/reducer'
+import { formatShortenedAddress } from 'src/components/ShortenedAddress'
+import {
+  AddressToDisplayNameType,
+  AddressToE164NumberType,
+  E164NumberToAddressType,
+} from 'src/identity/reducer'
 import { ContactMatches, RecipientVerificationStatus } from 'src/identity/types'
 import Logger from 'src/utils/Logger'
 
 const TAG = 'recipients/recipient'
 
-export enum RecipientKind {
-  MobileNumber = 'MobileNumber',
-  Contact = 'Contact',
-  QrCode = 'QrCode',
-  Address = 'Address',
+export type Recipient = {
+  name?: string
+  contactId?: string // unique ID given by phone OS
+  thumbnailPath?: string
+  displayNumber?: string
+} & ({ e164PhoneNumber: string } | { address: string })
```

Comment:
Hm are you sure? It should be an inclusive OR, which allows both of them to be present. Seems to work in the TS playground: https://www.typescriptlang.org/play?#code/C4TwDgpgBAShDGBLMiIDthQLxQN4FgAoKKNAQwFsIB+ALigGdgAnRNAcyJPgHsMz4wAJIATOoxZt2UAPQyoAVzSIAjguhCAIlHaIAbuigAjEFDAALPtADyAZS5Rg5hRSPlEAGwAKZJ+KasHA4iiAxgHmQgAHIuRhDM-pJBhAC+UABkUAAUuFAQAIwAbAAsXpZoEDGu8fQBUlBpAD54UGQiIswQDAy1SdIpAJRERLxoTFCdPXBIKOiYOLhtHV09UABEAAwAHhuFG2Qb8PAAZm3FZACcxYUATAAcFwCsZBAlT4UA7MUfRjdGIvkPscNu0PmsADQkAolMpWKpxZj0NbFe4AZgu6MxENIlAgSIAKl1gGsUsNCKMGDwPBAAHQeHjsLKTIaEIA

Reply:
You're right, it works. I just assumed the `|` operator was exclusive. The only negative consequence imo with your approach is that if you have a `Recipient` object you can't call `recipient.address` or `recipient.e164PhoneNumber`, you have to check first if the field exists like you're doing in `recipientHasAddress`:
```
export function recipientHasAddress(recipient: Recipient): recipient is AddressRecipient {
  return recipient && 'address' in recipient && !!recipient.address
}
```
which with the other implementation could be
```
export function recipientHasAddress(recipient: Recipient): recipient is AddressRecipient {
  return Boolean(recipient.address)
}
```
not the end of the world, since the 'ugliness' is just contained to a single method.

TIL 😄

Reply:
yes, unfortunately whenever I want to reference `address` or `e164phoneNumber` I always have to do `recipientHasAddress(recipient) && recipient.address`, whereas normally you can just reference `recipient.address` and let it resolve to undefined if it doesn't exist. Not sure how to get around this :/